language/compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -1 +1,1798 @@
-include!(concat!(env!("OUT_DIR"), "/syntax.rs"));
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::str::FromStr;
+use std::fmt;
+use codespan::{ByteIndex, Span};
+
+use crate::ast::{
+    ModuleDefinition, StructDefinition, Script, Program, ScriptOrModule,
+    FunctionBody, FunctionVisibility, ImportDefinition, ModuleName,
+    Block, Block_, Cmd, CopyableVal, CopyableVal_, Spanned, Kind, TypeVar,
+    Exp_, Exp, Var,  Var_, FunctionCall, FunctionCall_,
+    FunctionName, Builtin, Statement, IfElse, While, Loop, Type, Fields,
+    StructName, QualifiedStructIdent, Function, BinOp, ModuleIdent, QualifiedModuleIdent, UnaryOp,
+    LValue, LValue_, parse_field, Field_, TypeVar_, Function_, StructDefinition_,
+};
+use crate::lexer::*;
+use libra_types::{account_address::AccountAddress, byte_array::ByteArray};
+use hex;
+
+// FIXME: The following simplified version of ParseError copied from
+// lalrpop-util should be replaced.
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ParseError<L, T, E> {
+    InvalidToken { location: L },
+    UnrecognizedToken {
+        token: (L, T, L),
+        expected: Vec<String>,
+    },
+    User { error: E },
+}
+
+/// Format a list of expected tokens.
+fn fmt_expected(f: &mut fmt::Formatter, expected: &[String]) -> fmt::Result {
+    if !expected.is_empty() {
+        writeln!(f)?;
+        for (i, e) in expected.iter().enumerate() {
+            let sep = match i {
+                0 => "Expected one of",
+                _ if i < expected.len() - 1 => ",",
+                // Last expected message to be written
+                _ => " or",
+            };
+            write!(f, "{} {}", sep, e)?;
+        }
+    }
+    Ok(())
+}
+
+impl<L, T, E> fmt::Display for ParseError<L, T, E>
+where
+    L: fmt::Display,
+    T: fmt::Display,
+    E: fmt::Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        use self::ParseError::*;
+        match *self {
+            User { ref error } => write!(f, "{}", error),
+            InvalidToken { ref location } => write!(f, "Invalid token at {}",
+            location),
+            UnrecognizedToken {
+                token: (ref start, ref token, ref end),
+                ref expected,
+            } => {
+                write!(
+                    f,
+                    "Unrecognized token `{}` found at {}:{}",
+                    token, start, end
+                )?;
+                fmt_expected(f, expected)
+            }
+        }
+    }
+}
+
+fn consume_token<'input, 'builder>(tokens: &mut __Matcher<'input, 'builder>, tok: Tok)
+    -> Result<(), ParseError<usize, Token<'input>, failure::Error>>
+{
+    if (tokens.token.1).0 != tok {
```

Comment:
Do you need the parens here? If so, I'm surprised Clippy doesn't complain

Reply:
Yes, you do need the parens, but I've changed all the code to use the new lexer interface functions instead of accessing the tokens directly like that.